### PR TITLE
Implement xxx.Connection.CloseTimeout/ConnectTimeout in C#

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR implements xxx.Connection.ConnectTimeout and xxx.Connection.CloseTimeout in C#.

It also removes all implementation of the "connection timeouts" (superseded by idle timeout). And removes the old Connect/Close timeouts.

### CODE REVIEW

File: csharp/src/Ice/LocalException.cs [hunk 1]
```diff
@@ -1235,6 +1235,18 @@ public override string ice_id()
     }
 }
 
+/// <summary>
+/// This exception indicates that a connection was aborted by the idle check.
+/// </summary>
+public class ConnectionIdleException : LocalException
```
Comment (bernardnormier) [inline csharp/src/Ice/LocalException.cs:1241]: Renaming + no longer derives from TimeoutException.

File: csharp/src/Ice/Internal/OutgoingAsync.cs [hunk 2]
```diff
@@ -450,19 +446,6 @@ public override bool exception(Ice.Exception exc)
         }
     }
 
-    public override void cancelable(CancellationHandler handler)
-    {
-        if (proxy_.iceReference().getInvocationTimeout() == -2 && cachedConnection_ != null)
```
Comment (bernardnormier) [inline csharp/src/Ice/Internal/OutgoingAsync.cs:455]: Removed all the deprecated invocation timeout = -2 logic in C#.

File: csharp/src/Ice/Internal/DefaultsAndOverrides.cs [hunk 3]
```diff
@@ -58,46 +58,6 @@ internal DefaultsAndOverrides(Ice.Properties properties, Ice.Logger logger)
             overrideTimeoutValue = -1;
```
Comment (bernardnormier) [inline csharp/src/Ice/Internal/DefaultsAndOverrides.cs:58]: I think we should eliminate the "override timeout" in a follow-up PR (in C++ and C#).

File: csharp/src/Ice/ConnectionI.cs [hunk 4]
```diff
@@ -2874,17 +2791,10 @@ internal void completed(LocalException ex)
     private TraceLevels _traceLevels;
     private Ice.Internal.ThreadPool _threadPool;
 
-    private Ice.Internal.Timer _timer;
```
Comment (bernardnormier) [inline csharp/src/Ice/ConnectionI.cs:2877]: The ConnectionI code no longer uses our "traditional" Ice-C++-like Timer class.

**Review by pepone — APPROVED** [review]

**Review by externl — APPROVED** [review]

**Pull request merged**